forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt

```diff
@@ -0,0 +1,11 @@
+Name:Gilraen, Dunedain Protector
+ManaCost:2 W
+Types:Legendary Creature Human Noble
+PT:2/3
+A:AB$ ChangeZone | Cost$ 2 T | ValidTgts$ Creature.Other+YouCtrl | Origin$ Battlefield | Destination$ Exile | TgtPrompt$ Select another target creature you control | SubAbility$ DBReturn | SpellDescription$ Exile another target creature you control. You may return that card to the battlefield under its owner's control. If you don't, at the beginning of the next end step, return that card to the battlefield under its owner's control with a vigilance counter and a lifelink counter on it.
+SVar:DBReturn:DB$ ChangeZone | Defined$ Targeted | Origin$ All | Optional$ True | Destination$ Battlefield | RememberChanged$ True | SubAbility$ DBDelayedTrigger
+SVar:DBDelayedTrigger:DB$ DelayedTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card | ConditionCompare$ EQ0 | RememberObjects$ RememberedLKI | Mode$ Phase | Phase$ End Of Turn | Execute$ TrigReturn | SubAbility$ DBCleanup | SpellDescription$  If you don't, at the beginning of the next end step, return that card to the battlefield under its owner's control with a vigilance counter and a lifelink counter on it.
+SVar:TrigReturn:DB$ ChangeZone |  Defined$ DelayTriggerRememberedLKI | Origin$ Exile | Destination$ Battlefield | WithCountersType$ Vigilance,Lifelink | WithCountersAmount$ 1
```

Comment:
card untested imo

Reply:
Yeah forgot to note that the delayedtrigger isn't working. Any pointers to cards I should look at ?

Reply:
What doesn't work now?
(Except for the missing counters)

Reply:
it triggers at end of turn (if you chose to not return it immediately) but nothing gets returned

Reply:
you might want to draw a diagram of the effect chain for yourself and in what cases something would get remembered to understand what is happening

Reply:
got the delayedtrigger to work though not the etb counters. Does this need an additional VAR ?